internal/api/identity.go

```diff
@@ -57,6 +57,12 @@ func (a *API) DeleteIdentity(w http.ResponseWriter, r *http.Request) error {
 		if terr := tx.Destroy(identityToBeDeleted); terr != nil {
 			return internalServerError("Database error deleting identity").WithInternalError(terr)
 		}
+		if terr := user.UpdateUserEmail(tx); terr != nil {
+			if models.IsUniqueConstraintViolatedError(terr) {
+				return forbiddenError("Unable to unlink primary identity").WithInternalError(terr)
```

Comment:
open to suggestions on how we can make the error message clearer here, the idea is to inform the user that the primary identity cannot be unlinked because the other identities' emails are used as primary emails for other users already